test/src/compose/service.spec.ts

```diff
@@ -0,0 +1,1034 @@
+				BALENA_HOST_OS_VERSION: 'Resin OS 2.0.2',
+				BALENA_SERVICE_NAME: 'serviceName',
+				BALENA_APP_LOCK_PATH: '/tmp/balena/updates.lock',
+				BALENA_SERVICE_HANDOVER_COMPLETE_PATH: '/tmp/balena/handover-complete',
+				BALENA: '1',
+				USER: 'root',
+			});
+		});
+
+		it('returns the correct default bind mounts', async () => {
+			const s = await Service.fromComposeObject(
+				{
+					appId: '1234',
+					serviceName: 'foo',
+					releaseId: 2,
+					serviceId: 3,
+					imageId: 4,
+				},
+				{ appName: 'foo' } as any,
+			);
+			const binds = (Service as any).defaultBinds(s.appId, s.serviceName);
+			expect(binds).to.deep.equal([
+				'/tmp/balena-supervisor/services/1234/foo:/tmp/resin',
+				'/tmp/balena-supervisor/services/1234/foo:/tmp/balena',
+			]);
+		});
+
+		it('produces the correct port bindings and exposed ports', async () => {
+			const s = await Service.fromComposeObject(
+				{
+					appId: '1234',
+					serviceName: 'foo',
+					releaseId: 2,
+					serviceId: 3,
+					imageId: 4,
+					expose: [1000, '243/udp'],
+					ports: ['2344', '2345:2354', '2346:2367/udp'],
+				},
+				{
+					imageInfo: {
+						Config: {
+							ExposedPorts: {
+								'53/tcp': {},
+								'53/udp': {},
+								'2354/tcp': {},
+							},
+						},
+					},
+				} as any,
+			);
+
+			const ports = (s as any).generateExposeAndPorts();
+			expect(ports.portBindings).to.deep.equal({
+				'2344/tcp': [
+					{
+						HostIp: '',
+						HostPort: '2344',
+					},
+				],
+				'2354/tcp': [
+					{
+						HostIp: '',
+						HostPort: '2345',
+					},
+				],
+				'2367/udp': [
+					{
+						HostIp: '',
+						HostPort: '2346',
+					},
+				],
+			});
+			expect(ports.exposedPorts).to.deep.equal({
+				'1000/tcp': {},
+				'243/udp': {},
+				'2344/tcp': {},
+				'2354/tcp': {},
+				'2367/udp': {},
+				'53/tcp': {},
+				'53/udp': {},
+			});
+		});
+
+		it('correctly handles port ranges', async () => {
+			const s = await Service.fromComposeObject(
+				{
+					appId: '1234',
+					serviceName: 'foo',
+					releaseId: 2,
+					serviceId: 3,
+					imageId: 4,
+					expose: [1000, '243/udp'],
+					ports: ['1000-1003:2000-2003'],
+				},
+				{ appName: 'test' } as any,
+			);
+
+			const ports = (s as any).generateExposeAndPorts();
+			expect(ports.portBindings).to.deep.equal({
+				'2000/tcp': [
+					{
+						HostIp: '',
+						HostPort: '1000',
+					},
+				],
+				'2001/tcp': [
+					{
+						HostIp: '',
+						HostPort: '1001',
+					},
+				],
+				'2002/tcp': [
+					{
+						HostIp: '',
+						HostPort: '1002',
+					},
+				],
+				'2003/tcp': [
+					{
+						HostIp: '',
+						HostPort: '1003',
+					},
+				],
+			});
+
+			expect(ports.exposedPorts).to.deep.equal({
+				'1000/tcp': {},
+				'2000/tcp': {},
+				'2001/tcp': {},
+				'2002/tcp': {},
+				'2003/tcp': {},
+				'243/udp': {},
+			});
+		});
+
+		it('should correctly handle large port ranges', async function () {
+			this.timeout(60000);
```

Comment:
what's this for ?

Reply:
No idea, that was there before probably. I guess is a timeout for the test? Not sure why it's needed though